duckduckgo/Android
# Replace DL manager for file download

Task/Issue URL: 
* https://app.asana.com/0/488551667048375/1202527564835369/f
* https://app.asana.com/0/488551667048375/1202527564835371/f
* https://app.asana.com/0/0/1202527564835370/f

### Description
Use our own implementation of file downloads rather than relying on Android DL manager.
Also remove the code that force the traffic of our app to pass through the VPN in Android 12 devices.


Also, in subsequent PRs
* further interface/method/visibility clean ups will be done (no change in functionality)
* copy changes result of some other internal reviews
* (maybe) quick functionality wins now that we control the downloading process

This PR:
* makes the necessary changes to replace the DL manager
* cleans up some unused resource files associated to DL manager (may be more)

### Steps to test this PR
To test this PR run test cases as defined in [this asana task](https://app.asana.com/0/72649045549333/1202527564835393)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202527564835371
